agent/agent.go

```diff
@@ -237,6 +237,57 @@ func (a *Agent) GetRunningOutputPlugins() []string {
 	return res
 }
 
+type MapFieldSchema struct {
+	Value interface{}
+	Key   string
+}
+
+type ArrayFieldSchema struct {
+	Value  interface{}
+	Length int // Will be 0 if slice and array length if array
```

Comment:
Why not just have an array of interfaces, rather than an interface (that you know will be an array), with a length?

Reply:
We assumed that arrays will have the same element (value) type, and just wanted to know the length in case the plugin config is expecting a certain number of elements. Is that a bad assumption?

Reply:
So Value is essentially an array of interfaces, and you use the Length field to see the length of the array of interfaces? Sorry I'm a little confused. Why not just make it an array of interfaces at that point?

Reply:
Value describes the type/structure of a single element in the array. It's usually some sort of primitive (represented by Reflect.Type), but we declare it as a generic `interface{}` in case each array field is actually a map or another array that needs to be further traversed to find types more deeply. I'm not sure if this is an actual use case, so it may not be necessary to handle arrays of structs/maps/arrays. For example, maybe the plugin wants to define two specific sensors to measure from, and is storing them as an array of structs:
```
{"I2CSensors": [
  {"addr":1, "samplingRate":1000},
  {"addr":2, "samplingRate":1000}
]}
```
then the corresponding schema would be:
```
{"I2CSensors": {
  {"Length": 0, "Value":
    {"addr": int, "samplingRate": int}
]}
```

Reply:
I looked through the config file and it looks like there aren't any use cases that really match this implementation though -- most arrays are just of strings or numbers. We could simplify it if you'd like, we just wrote the function to be as general as possible so I think it's probably a little too confusing.

Reply:
Oh I see. So this isn't a wrapper around the whole array, it's a wrapper around individual elements within the array. Gotcha. This is okay!